pkg/services/sqlstore/datasource.go

```diff
@@ -71,6 +72,20 @@ func GetDataSources(query *models.GetDataSourcesQuery) error {
 	} else {
 		sess = x.Limit(query.DataSourceLimit, 0).Where("org_id=?", query.OrgId).Asc("name")
 	}
+
+	query.Result = make([]*models.DataSource, 0)
+	return sess.Find(&query.Result)
+}
+
+// GetDataSourcesByType returns all datasources for a given type
+func GetDataSourcesByType(query *models.GetDataSourcesByTypeQuery) error {
+	var sess *xorm.Session
+	sess = x.Asc("id")
+
+	if query.Type != "" {
```

Comment:
Maybe return an error if type is empty instead? If someone don't want to filter by type the `GetDataSourcesQuery` should be used instead.

Reply:
makes sense, changed it.